### PR TITLE
Extract guice bindings for custom Iceberg catalog

### PR DESCRIPTION
**Pull request by findinpath**
## Description

Allow tests to use a fully customized Iceberg catalog
that is not necessarily backed by a `FileHiveMetastore` instance.

## Non-technical explanation

Test infrastructure change

## Release notes

(x) This is not user-visible and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Review by alexjo2144 — APPROVED** [review]
Looks good to me. It's nice to be able to move some of that test only code out of IcebergCatalogModule.

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/MetastoreValidator.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.catalog;
+
+import com.google.inject.Inject;
+import io.trino.plugin.hive.metastore.cache.SharedHiveMetastoreCache;
+
+public class MetastoreValidator
```
Comment (findepi) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/MetastoreValidator.java:19]: The "Extract MetastoreValidator" commit adds a new class, but doesn't delete any existing code.
What am i missing?

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/file/TestingIcebergFileMetastoreCatalogModule.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    public TestingIcebergFileMetastoreCatalogModule(HiveMetastore metastore)
+    {
+        this.metastore = metastore;
+    }
+
+    @Override
+    protected void setup(Binder binder)
+    {
+        binder.bind(HiveMetastoreFactory.class).annotatedWith(RawHiveMetastoreFactory.class).toInstance(HiveMetastoreFactory.ofInstance(metastore));
+        binder.bind(MetastoreValidator.class).asEagerSingleton();
```
Comment (findepi) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/file/TestingIcebergFileMetastoreCatalogModule.java:42]: it doesn't need to do this IMO

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/file/TestingIcebergFileMetastoreCatalogModule.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import io.trino.plugin.iceberg.catalog.hms.TrinoHiveCatalogFactory;
+
+public class TestingIcebergFileMetastoreCatalogModule
+        extends AbstractConfigurationAwareModule
+{
+    private final HiveMetastore metastore;
+
+    public TestingIcebergFileMetastoreCatalogModule(HiveMetastore metastore)
+    {
+        this.metastore = metastore;
```
Comment (findepi) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/file/TestingIcebergFileMetastoreCatalogModule.java:35]: rnn

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/MetastoreValidator.java (hunk 1 above)
Comment (findinpath) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/MetastoreValidator.java:19]: I am actually missing attention.
Thank you for pointing out this problem.

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/file/TestingIcebergFileMetastoreCatalogModule.java (hunk 2 above)
Comment (findepi) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/file/TestingIcebergFileMetastoreCatalogModule.java]: In tests we don't need to validate that.

Comment (findepi) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/file/TestingIcebergFileMetastoreCatalogModule.java]: but fine, we can keep it

**Review by findepi — APPROVED** [review]

**Pull request merged**